### PR TITLE
express-serve-static-core: Make req.ip optional

### PR DESCRIPTION
**Pull request by aapolkovsky**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/expressjs/express/issues/4559#issuecomment-813071004

### CODE REVIEW

File: types/express-serve-static-core/index.d.ts [hunk 1]
```diff
@@ -562,11 +562,12 @@ export interface Request<
     secure: boolean;
 
     /**
-     * Return the remote address, or when
-     * "trust proxy" is `true` return
-     * the upstream addr.
+     * Return the remote address.
+     *
+     * If "trust proxy" is provided and none of the upstream addresses
+     * meet the conditions, then `undefined` will be returned.
```
Comment (JoseLion) [inline types/express-serve-static-core/index.d.ts:568]: When you say, "...and none of the upstream addresses meet the conditions", to what conditions are you referring? I think it'll be beneficial if we make that more explicit 🙂 

We should also add that the value may be `undefined` if the socket is destroyed.

**Review by JoseLion — COMMENTED** [review]
There's nothing in the Express.js [documentation](https://expressjs.com/en/4x/api.html#req.ip) (or the link provided) that explicitly mentions this property can be undefined.

However, I found an [issue comment](https://github.com/expressjs/express/issues/4559#issuecomment-813071004) in which a member of Express.js notes that `req.ip` is just an alias for [req.socket.remoteAddress](https://nodejs.org/api/net.html#net_socket_remoteaddress). Looking at this Node.js documentation, it does mention that:

> Value may be `undefined` if the socket is destroyed (for example, if the client disconnected).

I think this change makes complete sense, but we should change the documentation to reflect that 👍🏼

File: types/express-serve-static-core/index.d.ts (hunk 1 above)
Comment (aapolkovsky) [reply types/express-serve-static-core/index.d.ts]: > There's nothing in the Express.js [documentation](https://expressjs.com/en/4x/api.html#req.ip) (or the link provided) that explicitly mentions this property can be undefined.

Yes, I was wrong about the real reason why the ip might be undefined. I will remove this line and add comment about `req.socket.remoteAddress`.

**Review by JoseLion — APPROVED** [review]
LGTM! 🚀

**Comment by sandersn** [comment]
@typescript-bot got confused by the monorepo upgrade. This is ready to merge.

**Pull request merged**